server/app/com/xsn/explorer/services/BlockService.scala

```diff
@@ -60,6 +60,14 @@ class BlockService @Inject()(
     result.toFuture
   }
 
+  def getBlockHeader(height: Height): FutureApplicationResult[BlockHeader] = {
+    val result = for {
```

Comment:
You can replace the whole method with this line `blockDataHandler.getHeader(height)`